lib/src/op_store.rs

```diff
@@ -214,7 +214,10 @@ pub enum RemoteRefState {
 
 impl ContentHash for RemoteRefState {
     fn hash(&self, state: &mut impl digest::Update) {
-        (*self as u8).hash(state);
+        match self {
+            RemoteRefState::New => state.update(&0u32.to_le_bytes()),
+            RemoteRefState::Tracking => state.update(&1u32.to_le_bytes()),
+        }
```

Comment:
This would be simpler and more maintainable as `(*self as u32).to_le_bytes()).hash(state)`.

Reply:
This actually fails because the 4 byte array representing the u32 gets hashed using this implementation:

https://github.com/martinvonz/jj/blob/903f18acfd2365d4488eeea3425ab18615e23aa1/lib/src/content_hash.rs#L55-L62

I'm going to leave it as-is for this reason. It's a bit of a moot point though w/r/t maintainability since the goal is to replace the hand coded version with the proc macro.

Reply:
Closing the loop, with `state.update((*self as u32).to_le_bytes());` and a new non-unit variant you get a non-primitive cast error.

```
error[E0605]: non-primitive cast: `op_store::RemoteRefState` as `u32`
   --> lib/src/op_store.rs:218:23
    |
218 |         state.update(&(*self as u32).to_le_bytes());
    |                       ^^^^^^^^^^^^^^ an `as` expression can only be used to convert between primitive types or to coerce to a specific trait object
    ```

With the explicit match statement you get an error saying that the match doesn't cover all cases.

I'll add ContentHash for u32 in a separate commit and use it to simplify the derive macro. No need IMO to rewrite the manual impls since they're going to be deleted.